### PR TITLE
New Resource: `azurerm_api_management_identity_provider_aad`

### PR DESCRIPTION
**Pull request by aqche**
Partially addresses: #5044  

Adds the `azurerm_api_management_identity_provider_aad` resource.

```
=== RUN   TestAccAzureRMApiManagementIdentityProviderAAD_basic
=== PAUSE TestAccAzureRMApiManagementIdentityProviderAAD_basic
=== CONT  TestAccAzureRMApiManagementIdentityProviderAAD_basic
--- PASS: TestAccAzureRMApiManagementIdentityProviderAAD_basic (1989.79s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/apimanagement/tests 1989.82
```

### CODE REVIEW

File: azurerm/internal/services/apimanagement/tests/resource_arm_api_management_identity_provider_aad_test.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMApiManagementIdentityProviderAAD_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```
Comment (katbyte) [inline azurerm/internal/services/apimanagement/tests/resource_arm_api_management_identity_provider_aad_test.go]: could we make this 
```suggestion
  name     = "acctestRG-api-%d"
```

File: azurerm/internal/services/apimanagement/tests/resource_arm_api_management_identity_provider_aad_test.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  publisher_name      = "pub1"
+  publisher_email     = "pub1@email.com"
+  sku_name            = "Developer_1"
+}
+
+resource "azurerm_api_management_identity_provider_aad" "test" {
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  api_management_name = "${azurerm_api_management.test.name}"
+  client_id           = "aadclientid"
```
Comment (katbyte) [inline azurerm/internal/services/apimanagement/tests/resource_arm_api_management_identity_provider_aad_test.go]: I think client IDs are guids??

File: azurerm/internal/services/apimanagement/tests/resource_arm_api_management_identity_provider_aad_test.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2018-01-01/apimanagement"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMApiManagementIdentityProviderAAD_basic(t *testing.T) {
```
Comment (katbyte) [inline azurerm/internal/services/apimanagement/tests/resource_arm_api_management_identity_provider_aad_test.go]: Could we please add an update test where we change all properties that are not force new?

File: website/docs/r/api_management_identity_provider_aad.html.markdown [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+---
+
+# azurerm_api_management_identity_provider_aad
+
+Manages an API Management AAD Identity Provider.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "test" {
```
Comment (katbyte) [inline website/docs/r/api_management_identity_provider_aad.html.markdown]: Could we use example in the docs
```suggestion
resource "azurerm_resource_group" "example" {
```

File: website/docs/r/api_management_identity_provider_aad.html.markdown [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_api_management" "test" {
```
Comment (katbyte) [inline website/docs/r/api_management_identity_provider_aad.html.markdown]:
```suggestion
resource "azurerm_api_management" "example" {
```

File: website/docs/r/api_management_identity_provider_aad.html.markdown [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  publisher_name      = "My Company"
+  publisher_email     = "company@terraform.io"
+  sku_name            = "Developer_1"
+}
+
+resource "azurerm_api_management_identity_provider_aad" "test" {
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  api_management_name = "${azurerm_api_management.test.name}"
+  client_id           = "aadclientid"
```
Comment (katbyte) [inline website/docs/r/api_management_identity_provider_aad.html.markdown]: And a guid here? (i could be wrong about this)

File: website/docs/r/api_management_identity_provider_aad.html.markdown [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The ID of the API Management AAD Identity Provider.
+
+## Import
+
+API Management AAD Identity Provider can be imported using the `resource id`, e.g.
+
+```shell
+terraform import azurerm_api_management_identity_provider_aad.test /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.ApiManagement/service/instance1/identityProviders/aad
```
Comment (katbyte) [inline website/docs/r/api_management_identity_provider_aad.html.markdown]:
```suggestion
terraform import azurerm_api_management_identity_provider_aad.example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.ApiManagement/service/instance1/identityProviders/aad
```

File: azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_aad.go [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+				Sensitive:    true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"allowed_tenants": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
```
Comment (katbyte) [inline azurerm/internal/services/apimanagement/resource_arm_api_management_identity_provider_aad.go]: Could we validate these are GUIDs?

**Comment by aqche** [comment]
@katbyte thanks for the review! updated the PR based on your comments.

```
--- PASS: TestAccAzureRMApiManagementIdentityProviderAAD_basic (2333.44s)
--- PASS: TestAccAzureRMApiManagementIdentityProviderAAD_update (2409.81s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/apimanagement/tests 2414.165s
```

**Pull request merged**